### PR TITLE
refactor(google_adk): properly early-return in edge case

### PR DESCRIPTION
**Pull request by KowalskiThomas**
## Description

As title says. Without this `return`,  we would call/execute the wrapped function twice. 

**Note** not including a changelog entry because it's an edge case. Can add if you think it's worth it.

### CODE REVIEW

File: tests/contrib/google_adk/test_google_adk.py [hunk 1]
```diff
@@ -243,3 +246,28 @@ def failing_tool(query: str) -> dict:
     assert runner_span.get_tag("component") == "google_adk"
     assert runner_span.get_tag("google_adk.request.provider") == "google"
     assert runner_span.get_tag("google_adk.request.model") == "gemini-2.5-pro"
+
+
+@pytest.mark.asyncio
+async def test_call_tool_live_fallback_does_not_double_invoke_wrapped() -> None:
```
Comment (sabrenner) [inline tests/contrib/google_adk/test_google_adk.py:252]: any way to test this without explicitly importing the tracing function directly? i.e. did this come up from an escalation / personal testing?

Comment (KowalskiThomas) [reply tests/contrib/google_adk/test_google_adk.py:252]: > any way to test this without explicitly importing the tracing function directly?

I can look into it; possibly also we may just want to skip the regression test on that (after all there probably wouldn't have been a test if the bug had never been introduced to begin with). 

> did this come up from an escalation / personal testing?

This came from an automatic AI bug report.

Comment (sabrenner) [reply tests/contrib/google_adk/test_google_adk.py:252]: ok cool yeah if there's something that can be done then great but if not we can keep this test, i think better to have a regression test than not!

Comment (KowalskiThomas) [reply tests/contrib/google_adk/test_google_adk.py:252]: Let's keep it this way then :)

**Comment by KowalskiThomas** [comment]
@sabrenner I'll try to understand why the CI is failing and ping you again for review once ready!

**Comment by KowalskiThomas** [comment]
@sabrenner all good now if you can take another look 🙏

**Review by sabrenner — APPROVED** [review]
